File: sales-app/ClientApp/src/app/models/order.ts
import { Address } from "./address";

export interface Order {
  id: string;
  batchId: string;
  firstName: string;
  lastName: string;
  address: Address;
  email: string;
  item: string;
  quantity: number;
  totalCost: string;
  status: string;
  purchaseTimestamp: string;
  processingTimestamp: string;
  shippingTimestamp: string;
}

File: order-processing/src/main/java/com/sales/orderprocessing/models/Order.java
package com.sales.orderprocessing.models;

import java.util.Date;
import java.util.UUID;

public class Order {

    public String id;
    public String batchId;
    public String producerId;
    public String processingId;
    public String shippingId;
    public OrderStatus status;
    public String firstName;
    public String lastName;
    public Address address;
    public String email;
    public String item;
    public int quantity;
    public String totalCost;
    public Date purchaseTimestamp;
    public Date processingTimestamp;
    public Date shippingTimestamp;

    public void process() {
        processingId = UUID.randomUUID().toString();
        status = OrderStatus.Processing;
        processingTimestamp = new Date();
    }

    public void ship() {
        shippingId = UUID.randomUUID().toString();
        status = OrderStatus.Shipped;
        shippingTimestamp = new Date();
    }
}